src/main/java/brave/features/opentracing/BraveTracer.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright 2016-2017 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.features.opentracing;
+
+import brave.internal.Nullable;
+import brave.propagation.Propagation;
+import brave.propagation.TraceContext;
+import brave.propagation.TraceContextOrSamplingFlags;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.opentracing.propagation.Format;
+import io.opentracing.propagation.TextMap;
+
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class BraveTracer implements Tracer {
+
+    static final List<String> PROPAGATION_KEYS = Propagation.B3_STRING.keys();
+    static final TraceContext.Injector<TextMap> INJECTOR = Propagation.B3_STRING.injector(TextMap::put);
+    static final TraceContext.Extractor<TextMapView> EXTRACTOR = Propagation.B3_STRING.extractor(TextMapView::get);
+
+    private brave.Tracer brave4;
+
+    public static BraveTracer wrap(brave.Tracer brave4) {
+        if (brave4 == null) throw new NullPointerException("brave tracer == null");
+        return new BraveTracer(brave4);
+    }
+
+    private BraveTracer(brave.Tracer brave4) {
+        this.brave4 = brave4;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public SpanBuilder buildSpan(String operationName) {
+        return new BraveSpanBuilder(brave4, operationName);
+    }
+
+    /**
+     * {@inheritDoc}
```

Comment:
/**
  * Injects the underlying context using <a href="https://github.com/openzipkin/b3-propagation">B3 encoding</a>.
  */